src/app/common/hooks/use-scale-text.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useRef } from 'react';
+
+export function useScaleText() {
+  const ref = useRef<HTMLHeadingElement>(null);
+
+  useEffect(() => {
+    const el = ref.current;
+    if (el) {
+      const parentWidth = el.parentElement?.offsetWidth || 0;
+      const naturalWidth = el.scrollWidth;
+      const scale = parentWidth / naturalWidth;
+      const finalScale = Math.max(scale);
+      el.style.transform = finalScale < 1 ? `scale(${finalScale})` : 'none';
+    }
+  }, []);
+
+  return ref;
+}
```

Comment:
![image](https://github.com/user-attachments/assets/37ec2e8b-123f-4f32-8246-93d1ce0c844a)

new line at end of file to avoid this